cmd/server/handlers/handlemetric/service.go

```diff
@@ -76,17 +85,71 @@ func updateMetricByRequestBody(body payload.Metrics) (string, *UpdateMetricError
 		if body.Value == nil {
 			return "", BadRequestError
 		}
-		metrics.MeStore.SetGauge(body.ID, metrics.Gauge(*body.Value))
+		err := metrics.MeStore.SetGauge(body.ID, metrics.Gauge(*body.Value))
+		if err != nil {
+			//log.Println(err)
+			return "", &UpdateMetricError{err, http.StatusInternalServerError}
+		}
 		responseMessage = fmt.Sprintf("metric %s successfully set", body.ID)
 	case metrics.TypeCounter:
 		if body.Delta == nil {
 			return "", BadRequestError
 		}
-		metrics.MeStore.AddCounter(body.ID, metrics.Counter(*body.Delta))
+		err := metrics.MeStore.AddCounter(body.ID, metrics.Counter(*body.Delta))
+		if err != nil {
+			//log.Println(err)
+			return "", &UpdateMetricError{err, http.StatusInternalServerError}
+		}
 		responseMessage = fmt.Sprintf("metric %s successfully add", body.ID)
 	default:
 		return "", InvalidMetricTypeError
 	}
 
 	return responseMessage, nil
 }
+
+func updateMetricsByRequestBody(bodies []payload.Metrics) (string, *UpdateMetricError) {
```

Comment:
Не стоит возвращать конкретный тип ошибки. Стоит реализовать кастомную ошибку так, чтобы она реализовывала интерфейс error и возвращать его, а на вызывающей стороне можно будет использовать errors.As(...) для проверки.
```suggestion
func updateMetricsByRequestBody(bodies []payload.Metrics) (string, error) {
```

Reply:
Сделал возвращение ошибки, а в вызывающей функции проверяю через errors.As